src/utils/vaultMath.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  floorDivideBy,
+  floorMultiplyBy,
+} from '@agoric/zoe/src/contractSupport';
+import { AmountMath } from '@agoric/ertp';
+import { Ratio } from 'store/vaults';
+
+export const computeToReceive = (
+  priceRate: Ratio,
+  collateralizationRatio: Ratio,
+  toLock: bigint,
+  defaultCollateralization: Ratio,
+) => {
+  const collateralizationRatioOrDefault =
+    collateralizationRatio.numerator.value === 0n
+      ? defaultCollateralization
+      : collateralizationRatio;
+
+  const lockedPrice = floorMultiplyBy(
+    AmountMath.make(priceRate.denominator.brand, toLock),
+    priceRate,
+  );
+
+  return floorDivideBy(lockedPrice, collateralizationRatioOrDefault).value;
+};
+
+export const computeToLock = (
+  priceRate: Ratio,
+  collateralizationRatio: Ratio,
+  toReceive: bigint,
+  defaultCollateralization: Ratio,
+) => {
+  const collateralizationRatioOrDefault =
+    collateralizationRatio.numerator.value === 0n
+      ? defaultCollateralization
+      : collateralizationRatio;
+
+  const receiveMargin = floorMultiplyBy(
+    AmountMath.make(priceRate.numerator.brand, toReceive),
+    collateralizationRatioOrDefault,
+  );
+
+  return floorDivideBy(receiveMargin, priceRate).value;
+};
+
+/**
+ * Returns negative if left < right, 0 if left = right,
+ * positive if left < right.
+ */
+export const compareRatios = (left: Ratio, right: Ratio) => {
```

Comment:
I'm surprised contractSupport doesn't have this. We do have `ratioGTE` but that's boolean. 

a ratio comparator definitely belongs in contractSupport. not a release blocker but please include a TODO here to move it.

Reply:
Oops I didn't know about that, that suits our needs perfectly fine.